### PR TITLE
Feature Highlights: add highlight to toolbar button and add upgrade modal

### PR DESCRIPTION
**Pull request by AgnesToulet**
**What this PR does / why we need it**:
- Enable highlighting toolbar icons with a green dot
- Add an `UpgradeModal` component (heavily relying on the `UpgradeBox` component to keep the same design)

Used by enterprise to add highlights for specific features, but can have other use cases in the future.

**Special notes for your reviewer**:
Screenshot:
image

### CODE REVIEW

File: packages/grafana-ui/src/components/Button/ToolbarButton.tsx [hunk 1]
```diff
@@ -29,6 +29,8 @@ export interface Props extends ButtonHTMLAttributes<HTMLButtonElement> {
   variant?: ToolbarButtonVariant;
   /** Hide any children and only show icon */
   iconOnly?: boolean;
+  /** Show highlight dit */
```
Comment (Clarity-89) [inline packages/grafana-ui/src/components/Button/ToolbarButton.tsx:32]: dit?

File: packages/grafana-ui/src/components/Button/ToolbarButton.tsx [hunk 2]
```diff
@@ -84,6 +87,7 @@ export const ToolbarButton = forwardRef<HTMLButtonElement, Props>(
         {children && !iconOnly && <div className={contentStyles}>{children}</div>}
         {isOpen === false && <Icon name="angle-down" />}
         {isOpen === true && <Icon name="angle-up" />}
+        {isHighlighted === true && <div className={styles.highlight} />}
```
Comment (Clarity-89) [inline packages/grafana-ui/src/components/Button/ToolbarButton.tsx:90]:
```suggestion
        {isHighlighted && <div className={styles.highlight} />}
```
`isHighlighted` can only be a boolean or undefined so this check can be simplified.

**Review by Clarity-89 — COMMENTED** [review]
Looks good, the only thing I'm not sure about is the modal design, can't we put the upgrade box inside the usual modal and remove the upgrade button from the box? Then it'd look like in designs:
Screenshot 2022-01-31 at 13 38 41
Another benefit of this approach is that we wouldn't need to make changes to the modal component to make the title optional. Plus the title could provide a useful info about the feature name.

**Review by axelavargas — APPROVED** [review]
Great job @AgnesToulet 🚀 , I reviewed the changes related to our code owners and looks good :)

**Pull request merged**